api/test/unit/controllers/test_search_controller.py

```diff
@@ -826,12 +824,69 @@ def _delete_all_results_but_first(_, __, results, ___):
     assert "Nesting threshold breached" in caplog.text
 
 
-def test_get_excluded_providers_query_returns_None_when_no_provider_is_excluded():
-    assert search_controller.get_excluded_providers_query() is None
+@pytest.mark.django_db
+@pytest.mark.parametrize("cache_name", ["django_cache", "unreachable_django_cache"])
+@pytest.mark.parametrize(
+    "excluded_count, result",
+    [(2, Terms(provider=["provider1", "provider2"])), (0, None)],
+)
+def test_get_excluded_providers_query_returns_excluded(
+    excluded_count, result, cache_name, request, caplog
+):
+    cache = request.getfixturevalue(cache_name)
```

Comment:
TIL! That's cool!

Reply:
It is, there's a plugin for pytest that provides a clearer syntax than `request.getfixturevalue` but this is clear and less magic imo.

Reply:
Agreed! This is super neat!